gnucash/gnome/gnc-plugin-page-account-tree.c

```diff
@@ -1499,21 +1500,51 @@ gnc_plugin_page_account_tree_cmd_delete_account (GtkAction *action, GncPluginPag
          * Note that one effect of the modal dialog is preventing
          * the account selectors from being repopulated.
          */
-        response = gtk_dialog_run(GTK_DIALOG(dialog));
-        if (GTK_RESPONSE_ACCEPT != response)
+        account_currency = gnc_account_get_currency_or_parent (account);
+        // Ideally we should check all subaccount currencies, and verify they're all identical. If not, it's not advisable
+        // to move subaccount currencies to a new account.
+        while (1)
         {
-            /* Account deletion is cancelled, so clean up and return. */
-            gtk_widget_destroy(dialog);
-            g_list_free(filter);
-            g_free(acct_name);
-            return;
+            gboolean ta_match = TRUE;
+            response = gtk_dialog_run(GTK_DIALOG(dialog));
+            if (GTK_RESPONSE_ACCEPT != response)
+            {
+                /* Account deletion is cancelled, so clean up and return. */
+                gtk_widget_destroy(dialog);
+                g_list_free(filter);
+                g_free(acct_name);
+                return;
+            }
+            if (trans_mas && gtk_widget_is_sensitive(trans_mas))
+            {
+                ta = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(trans_mas));
+                ta_match = gnc_account_get_currency_or_parent (ta) == account_currency;
+            }
+            if (sa_mas && gtk_widget_is_sensitive(sa_mas))
+                saa = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(sa_mas));
+            if (sa_trans_mas && gtk_widget_is_sensitive(sa_trans_mas))
+                sta = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(sa_trans_mas));
```

Comment:
There's absolutely no point fo doing this inside the loop. Even given that the loop will seldom run more than once it only makes the code harder to read to include them here.

Reply:
mmm, if that's not in the while(), the user has no chance to change the account or cancel.

Reply:
True about the subaccount choices. I think that `sa_trans_mas` is anyway misguided. It's unlikely that it would make sense to put all of the transactions from all subaccounts into a single account. It would be OK if it were restricted to a single account but that would also need the commodity check.

Good that you brought up cancellation: You don't handle it so cancelling will just re-run the dialog. That would be frustrating.

Reply:
Actually, no, it's not true, I was right in the first place. I guess you were thinking of reading `sa_mas` and `sa_trans_mas` before entering the loop and I was thinking of after. Only the values on the last run of the dialog matter.
*But* `sa_trans_mas` should either be removed or should only be enabled if there's only one subaccount, and if it's enabled it should also get the commodity-equal check.

Reply:
But how can I check the match of the currency if I don't read the selected account in the loop? I"m not understanding... 
So, should I change the behavior so we only allow moving sub-account transactions is there's only 1 sub-account? You don't expect people getting annoyed at this "regression"?

Reply:
> But how can I check the match of the currency if I don't read the selected account in the loop? I"m not understanding...

This is only about `sa_mas` and `sa_trans_mas`. Only `trans_mas` is significant for the loop as it's written, though `sa_trans_mas` either needs the same treatment or it needs to be removed.

> So, should I change the behavior so we only allow moving sub-account transactions is there's only 1 sub-account? You don't expect people getting annoyed at this "regression"?

I don't expect that anyone would notice because it's such a spectacularly stupid thing to do that nobody would try it. I think there would be whining about the  more general reduction in features I suggested in another comment, disallowing deleting an account that has any subaccounts, but I do think that's the best approach. It doesn't prevent the user from shuffling their CoA at will, but it does greatly reduce the opportunity for them to blow up their book while doing so.